### DISCUSSION TITLE
Object Detection for webCam

### DISCUSSION THREAD

**Discussion by hpssjellis** [huggingface/huggingface.js#90] [Q&A] [answered]
@coyotte508 I am starting to understand how huggingFaceJS works, which completely makes sense for text, audio and image classification. As far as I can tell each image/audio/text is uploaded to huggingface and the detection is done on the cloud and the reply is sent back to the webpage. Totally makes sense, however for webCam classification is there any method to load a smaller model (< 100 MB) onto the webpage and do the analysis client-side?

I assume the answer is "no".

**Comment by coyotte508** [comment]
@hpssjellis no, but see https://github.com/huggingface/huggingface.js/issues/82 where this is discussed

**Comment by hpssjellis** [comment]
That is OK, my background is in TensorflowJS and [edgeImpulse.com](https://www.edgeimpulse.com/) . It is great to have lots of webML options. TFJS has an advantage for client-side, huggingface is super easy to load data onto huge models. I am sure OpenAI has a place as well.

**Comment by hpssjellis** [comment, marked as answer]
You are going to like this @coyotte508 and @radames 

[hug08-webcam.html](https://hpssjellis.github.io/my-examples-of-huggingfacejs/public/hug08-webcam.html)


Thanks so much for making huggingFaceJS available to the public!

**Reply by coyotte508** [reply]
Thanks @hpssjellis ! 

We're probably going to reuse some of your examples (with credit + links) if that's ok in our doc!

**Reply by radames** [reply]
@hpssjellis I like your idea to store the token on localStorage! One tip, you could use a classless css framework to make your examples mobile and dark themed. e.g https://simplecss.org/ or pick any here https://github.com/dbohdan/classless-css

**Reply by hpssjellis** [reply]
@radames and @coyotte508 use what you want. I do all my code for in person teaching. I dont spend a ton of time making it perfect. I am making more examples.
